src/span.js

```diff
@@ -137,24 +137,51 @@ export default class Span {
     }
 
     /**
+     * Checks whether or not a span can be written to.
+     *
+     * @return {boolean} - The decision about whether this span can be written to.
+     **/
+    _isWriteable(): boolean {
+        // A span can NOT be written to if it:
+        // 1.) is finished.
+        // 2.) has been finalized, and is not sampled
+        return  this._duration === undefined &&
+                (!this._spanContext.samplingFinalized ||
+                this._spanContext.isSampled());
+    }
+
+    /**
      * Sets the operation name on this given span.
      *
      * @param {string} name - The name to use for setting a span's operation name.
      * @return {Span} - returns this span.
      **/
     setOperationName(operationName: string): Span {
+        if (!this._isWriteable()) {
```

Comment:
I am not sure about that. Say the span is not sampled but finalized because the sampling was inherited from the parent. Why shouldn't we still be able to override the operation name?